subprojects/gradle/buildchecks/src/main/kotlin/com/avito/android/plugin/build_param_check/incremental_check/IncrementalKaptTask.kt

```diff
@@ -0,0 +1,47 @@
+package com.avito.android.plugin.build_param_check.incremental_check
+
+import com.avito.android.plugin.build_param_check.CheckResult
+import com.avito.android.plugin.build_param_check.CheckTaskWithMode
+import com.avito.kotlin.dsl.getBooleanProperty
+import org.gradle.api.Project
+import org.gradle.api.tasks.TaskAction
+
+internal abstract class IncrementalKaptTask : CheckTaskWithMode() {
+
+    @TaskAction
+    fun check() {
+        val incrementalKaptEnabled = project.getBooleanProperty("kapt.incremental.apt", default = false)
+        if (incrementalKaptEnabled) {
+            checkAnnotationProcessors()
+        }
+    }
+
+    private fun checkAnnotationProcessors() {
```

Comment:
not a blocker

How long does it take on avito project?
It seems that we can check projects in parallel.

Reply:
I've run 5 tests 5 times and the maximum time I've got is about 30ms.